swan-lake/resources/featured-scenarios/write-a-restful-api-with-ballerina.md

```diff
@@ -127,103 +187,75 @@ service /covid/status on new http:Listener(9000) {
-- The body of the response is of type `ErrorMsg`, which simply has a string field named `errmsg`. Based on the need, users can have any data type for their response body.
-- Ballerina has a defined set of types for each HTTP status code. This allows you to write services in a type-oriented way, which in turn is helpful when it comes to tooling and generating OpenAPI specifications for HTTP services. 
+- It is chosen to accept the entire payload or send back an error.
+- This resource has an argument named `covidEntries`, which means the resource expects a payload with the `CovidEntry[]` type. Two types of records, `CovidEntry[]` and `ConflictingIsoCodesError`, will be used as the return types.
 
 ## Implement the second endpoint
 
 The second endpoint has only one resource to get COVID-19 data filtered by the ISO code.
 
 ### Create the resource of the second endpoint
 
-To create the resource of the second endpoin, add the code below to the API template file (i.e., `service.bal`).
+Similar to how you created the [second resource of the first endpoint](#create-the-second-resource-to-add-data), create the resource of the second endpoint below using the diagram view in VS Code.
+
+>**Note:** Define an HTTP resource that allows the `GET` operation on the resource path `countries` and accepts the `iso_code` path parameter. Define and use `InvalidIsoCodeError` along with `CovidEntry[]` as the response types.
+![Create second GET resource](/learn/images/featured-scenarios/write-a-restful-api-with-ballerina/create-second-get-resource.gif)
+
+Implement the logic of this `GET` resource function with the code below.
 
 ```ballerina
 resource function get countries/[string iso_code]() returns CovidEntry|InvalidIsoCodeError {
     CovidEntry? covidEntry = covidTable[iso_code];
     if covidEntry is () {
         return {
-            body: {
-                errmsg: string `Invalid ISO Code: ${iso_code}`
-            }
+            body: string `Invalid ISO Code: ${iso_code}`
         };
     }
     return covidEntry;
 }
 ```
 
 In this code:
-- This resource is a bit more different than the first two resources. As explained earlier, resource methods have accessors.
-- In addition, it also supports hierarchical paths making it ideal for implementing RESTful APIs. Hierarchical paths can have path params.
--  In this case, `iso_code` is used as the path param, which in turn, becomes a string variable.
-
-#### Define the error record
-
-To define the `InvalidIsoCodeError` record, add the code below to the API template file (i.e., `service.bal`).
-
-```ballerina
-public type InvalidIsoCodeError record {|
-    *http:NotFound;
-    ErrorMsg body;
-|};
-```
-
-In this code:
-- As in the previous example, this resource also includes its own return types. However, the basic principle behind them is as the previous example. 
+- This resource is different from the first two resources. As explained earlier, resource methods have accessors.
```

Comment:
What does this means? The two sentences does not make sense. It is different since it has a path parameter?

Reply:
Updated as follows:

In this code:
- Resource methods also support hierarchical paths, making it ideal for implementing RESTful APIs. Hierarchical paths can have path parameters.
- This resource is different from the first two resources because it has a path parameter.
- In this case, `iso_code` is used as the path param, which, in turn, becomes a `string` variable.